packages/portfolio-contract/src/resolver/resolver.exo.ts

```diff
@@ -0,0 +1,253 @@
+              return;
+          }
+        },
+      },
+      debug: {
+        getPendingTransactions() {
+          return [...cctpTransactionRegistry.keys()];
+        },
+      },
+    },
+  );
+};
+
+harden(prepareCCTPResolver);
+
+/**
+ * Prepare CCTP Resolver Invitation Makers.
+ *
+ * @param zone - Durable storage zone
+ * @param zcf - Zoe Contract Facet for creating invitations
+ * @param resolver - The resolver instance to use
+ * @returns A function to create resolver invitation makers
+ */
+export const prepareResolverInvitationMakers = (
+  zone: Zone,
+  zcf: ZCF,
+  resolver: CCTPResolverServiceFacet,
+) => {
+  const resolverInvitationMakersZone = zone.subZone('ResolverInvitationMakers');
+
+  // Prepare durable settlement handler
+  const prepareSettlementHandler = resolverInvitationMakersZone.exoClass(
+    'SettlementHandler',
+    M.interface('SettlementHandler', {
+      handle: M.call(M.any(), M.any()).returns(M.promise()),
+    }),
+    () => ({ resolver }),
+    {
+      async handle(seat: ZCFSeat, offerArgs: CCTPSettlementOfferArgs) {
+        const { resolver } = this.state;
+        const offerArgsShape = M.splitRecord(
+          {
+            txDetails: CCTPTransactionDetailsShape,
+            remoteAxelarChain: M.string(), // CaipChainId format
+          },
+          {},
+          {},
+        );
+        mustMatch(offerArgs, offerArgsShape);
+        const { txDetails, remoteAxelarChain } = offerArgs as any;
+
+        trace('CCTP transaction settlement:', {
+          amount: txDetails.amount,
+          remoteAddress: txDetails.remoteAddress,
+          status: txDetails.status,
+          remoteAxelarChain,
+        });
+
+        seat.exit();
+
+        try {
+          resolver.settleCCTPTransaction(
+            remoteAxelarChain,
+            txDetails.remoteAddress,
+            txDetails.amount,
+            txDetails.status,
+          );
+
+          return harden({
+            success: true,
+            message: 'CCTP transaction settlement processed',
+            key: `${remoteAxelarChain}:${txDetails.remoteAddress}:${txDetails.amount}`,
+            txDetails,
+            remoteAxelarChain,
+          });
+        } catch (error) {
+          return harden({
+            success: false,
+            message: error.message || 'CCTP transaction settlement failed',
+            key: `${remoteAxelarChain}:${txDetails.remoteAddress}:${txDetails.amount}`,
+            txDetails,
+            remoteAxelarChain,
+          });
+        }
+      },
+    },
+  );
+
+  const ResolverInvitationMakersI = M.interface('ResolverInvitationMakers', {
+    makeSettleCCTPTransactionInvitation: M.call().returns(M.promise()),
+  });
+
+  return resolverInvitationMakersZone.exoClass(
+    'ResolverInvitationMakers',
+    ResolverInvitationMakersI,
+    () => ({ settlementHandler: prepareSettlementHandler() }),
+    {
+      makeSettleCCTPTransactionInvitation() {
```

Comment:
To follow the invitationMakers pattern:

```suggestion
      SettleCCTPTransaction() {
```

**design** question (external API)